hazelcast/src/main/java/com/hazelcast/internal/blackbox/Blackbox.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.blackbox;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * The Blackbox is responsible for recording all kinds of Hazelcast/JVM specific information to
+ * help out with all kinds of issues
+ *
+ * TODO:
+ * - number of invocations
+ * - number of failed invocation
+ * - number of retried invocations
+ * - back pressure information
+ * - runtime.availableMemory=NA  <<=== sensor doesn't exist.
+ *
+ * - tcp total number of created client connections
+ * - cluster: number of heart beats send
+ * - cluster: number of heart beats received
+ * - io number of rebalances
+ * - number per type of packet send
+ * - number per type of packet received
+ */
+public interface Blackbox {
```

Comment:
Also many methods below are lacking in documentation, including conistency of Javadoc with method signature.